src/wallet/wallet.cpp

```diff
@@ -1413,6 +1413,13 @@ int CWallet::GetRealOutpointCoinJoinRounds(const COutPoint& outpoint, int nRound
         }
     }
 
+    // make sure we spent all of it with 0 fee, reset to 0 rounds otherwise
+    if (wtx->GetDebit(ISMINE_SPENDABLE) != wtx->GetCredit(ISMINE_SPENDABLE)) {
```

Comment:
should we have a test for it? I am not sure how to reproduce this issue.

Test is also useful to avoid a regression in future

Reply:
>I am not sure how to reproduce this issue.

Go to CoinJoin tab -> Inputs... -> Select 10 same mid-size inputs (e.g. 1.00001000) and 1 the smallest input (e.g. 0.00100001) to pay the fee -> Close the dialog and pay to some address 10x"mid-size" i.e. the next denom amount (e.g. 10.00010000). Check Inputs... again and you'll se that the most recent input (0 confirmations) has non-zero rounds. Apply the patch and check the same input again, should be 0 now.

Reply:
can you make an unit test for this scenario to avoid regression?